core/src/test/java/org/apache/accumulo/core/file/rfile/RFileTest.java

```diff
@@ -304,10 +304,10 @@ public void openReader(boolean cfsi) throws IOException {
       LruBlockCache indexCache = (LruBlockCache) manager.getBlockCache(CacheType.INDEX);
       LruBlockCache dataCache = (LruBlockCache) manager.getBlockCache(CacheType.DATA);
 
-      CachableBlockFile.Reader _cbr = new CachableBlockFile.Reader("source-1", in, fileLength, conf,
-          dataCache, indexCache,
-          CryptoServiceFactory.newInstance(accumuloConfiguration, ClassloaderType.JAVA));
-      reader = new RFile.Reader(_cbr);
+      CachableBuilder cb = new CachableBuilder().cacheId("source-1").input(in).length(fileLength)
```

Comment:
It looks like you are always creating a reader.  Could you have your builder just create one (instead of CachableBuilder) ? Like the following:

```java
      reader = RFile.reader().cacheId("source-1").input(in).length(fileLength)
          .conf(conf).data(dataCache).index(indexCache).cryptoService(
              CryptoServiceFactory.newInstance(accumuloConfiguration, ClassloaderType.JAVA));
```

Reply:
Yeah I should be able to do that.  There are a few places where the CachableBlockFile reader are used without the RFile but they might just be for tests.

Reply:
So I figured out that Summaries use just the BCFile reader and not the entire RFile.  I think this was an optimization done for reading summary data.  So for now, we still need it separate from RFile.

Reply:
Ok sounds good